server/controllers/oauth.py

```diff
@@ -91,6 +112,13 @@ def authorize(*args, **kwargs):
     confirm = request.form.get('confirm', 'no')
     return confirm == 'yes'
 
+@oauth.route('/oauth/code')
+def oauth_code():
```

Comment:
For what it's worth - Google's code view is a POST.  Refreshes trigger a resubmission warning and generate a new code. 

You can try it out with
> gcloud auth login --no-launch-browser